Frameworks/CoreText/DWriteWrapper_CoreText.mm

```diff
@@ -522,4 +523,56 @@ HRESULT STDMETHODCALLTYPE GetPixelsPerDip(_In_opt_ void* clientDrawingContext, _
     }
 
     return frame;
+}
+
+static CGSize _DWriteGetSize(CFAttributedStringRef string, CFRange range, CGSize maxSize, CFRange* fitRange) {
+    CGSize ret = CGSizeZero;
+    if (range.length == 0L) {
+        range.length = CFAttributedStringGetLength(string);
+    }
+
+    if (!string || range.length <= 0L) {
+        return ret;
+    }
+
+    ComPtr<IDWriteTextLayout> textLayout;
+    if (FAILED(__DWriteTextLayoutCreate(string, range, { CGPointZero, maxSize }, &textLayout))) {
+        return ret;
+    }
+
+    DWRITE_TEXT_METRICS textMetrics;
+    if (FAILED(textLayout->GetMetrics(&textMetrics))) {
+        return ret;
+    }
+
+    // TODO:: find more precise value than 1.0 to increase width by to fully enclose frame
+    ret.width = std::min(maxSize.width, textMetrics.widthIncludingTrailingWhitespace + 1.0f);
+    ret.height = std::min(maxSize.height, textMetrics.height);
+
+    if (fitRange) {
+        *fitRange = {};
+        uint32_t lineCount = 0;
+
+        // Ignore return value, we only want the lineCount for now
+        textLayout->GetLineMetrics(nullptr, 0, &lineCount);
```

Comment:
safer to check for that exact case (also good for unwanted behavior)
Given the implementation and documentation only mention it for the specific error above.
e.g someone decides to change the LINE_METRICS to and invalid state, this would help catch the error faster.

---
In reply to: [93510430](https://github.com/Microsoft/WinObjC/pull/1603#discussion_r93510430) [](ancestors = 93510430,93509971,93508458,93508293)